cmd/mesh/manifest-generate_test.go

```diff
@@ -156,12 +194,30 @@ func runTestGroup(t *testing.T, tests testGroup) {
 	}
 }
 
-// runManifestGenerate runs the manifest generate command. If flags is not set, passes the given path as a -f flag,
-// otherwise flags is passed to the command verbatim. Both path and flags should not be simultaneously set.
+// runManifestGenerate runs the manifest generate command. If path is set, passes the given path as a -f flag,
+// flags is passed to the command verbatim. If you set both flags and path, make sure to not use -f in flags.
 func runManifestGenerate(path, flags string) (string, error) {
-	args := "manifest generate " + flags
-	if flags == "" {
-		args += "-f " + path
+	args := "manifest generate"
+	if path != "" {
+		args += " -f " + path
+	}
+	if flags != "" {
+		args += " " + flags
 	}
 	return runCommand(args)
 }
+
+func createTempDirOrFail(t *testing.T, prefix string) string {
```

Comment:
It's ok as is but in general it's best to avoid passing testing.T into helper functions, because it makes the execution of the code harder to reason about. It's best to stick with the pattern or returning error - even though I get that here the intention was to avoid cluttering the main function.